Classes/PowerReportListControl.lua

```diff
@@ -57,22 +58,33 @@ function PowerReportListClass:ReSort(colIndex)
 			end
 			return a.pathDist < b.pathDist
 		end)
+	elseif colIndex == 5 then
+		t_sort(self.list, function (a,b)
+			if a.pathPower == b.pathPower and type(a.pathDist) == "number" and type(b.pathDist) == "number"then
+				return a.pathDist < b.pathDist
+			end
+			return a.pathPower > b.pathPower
+		end)
 	end
 end
 
 function PowerReportListClass:ReList()
+	if not next(self.originalList) then
+		return { }
+	end
 	local filteredList = { }
 	local iterate = 1
 	local insert = true
 
 	while(true) do
 
-		if (self.originalList[iterate].power <= 0) then
-			insert = false
-		end
+		insert = self.originalList[iterate].power > 0
 		if (not self.showClusters) and (self.originalList[iterate].pathDist == "Cluster") then
 			insert = false
 		end
+		if self.allocated and self.originalList[iterate].pathDist ~= "Cluster" then
+			insert = self.originalList[iterate].allocated and (self.originalList[iterate].pathDist <= (self.pathLength or 100))
```

Comment:
This isn't the doing of this PR, but the `pathDist` attribute is a bit of a mess. It can either assume the normal distance if there is a path, or some magic values (100, 1000, "Anoint", "Cluster"), depending on the circumstances. I'd recommend refactoring "Anoint" and "Cluster" to yet another two magic numbers to avoid having to check for the type of the attribute and to decouple computation with and presentation of these magic numbers, as well as documenting them. However, that's out of scope for this PR.

Reply:
Created a new issue in #2396